Native_UWP_V2/MainPage.xaml.cs

```diff
@@ -94,41 +88,36 @@ private async void CallGraphButton_Click(object sender, RoutedEventArgs e)
 
             if (authResult != null)
             {
-                var content = await GetHttpContentWithToken(graphAPIEndpoint, authResult.AccessToken).ConfigureAwait(false);
+                var graphServiceClient = GetGraphServiceClient(authResult.AccessToken);
+
+                // Call the /me endpoint of Graph
+                User graphUser = await graphServiceClient.Me.Request().GetAsync();
 
                 // Go back to the UI thread to make changes to the UI
                 await Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>
                 {
-                    ResultText.Text = content;
+                    ResultText.Text = "Display Name: " + graphUser.DisplayName + "\nBusiness Phone: " + graphUser.BusinessPhones.FirstOrDefault()
+                                      + "\nGiven Name: " + graphUser.GivenName + "\nid: " + graphUser.Id
+                                      + "\nUser Principal Name: " + graphUser.UserPrincipalName;
                     DisplayBasicTokenInfo(authResult);
                     this.SignOutButton.Visibility = Visibility.Visible;
                 });
             }
         }
 
         /// <summary>
-        /// Perform an HTTP GET request to a URL using an HTTP Authorization header
+        /// Instantiating GraphServiceClient using the access token.
         /// </summary>
-        /// <param name="url">The URL</param>
         /// <param name="token">The token</param>
-        /// <returns>String containing the results of the GET operation</returns>
-        private async Task<string> GetHttpContentWithToken(string url, string token)
+        /// <returns>GraphServiceClient</returns>
+        private GraphServiceClient GetGraphServiceClient(string token)
```

Comment:
This is not showing the right pattern, as this app will be usable only one hour, and then there will be an error that users won't understand.

Also if you want to use the GraphClient, also use the Auth adapters.
cc: @darrelmiller

Reply:
Updated implementation as discussed for Public client application.